packages/csharp/PortingAssistant/PortingAssistant.Api/Application.cs

```diff
@@ -109,6 +111,21 @@ public void SetupConnection(bool console = false)
                     };
                 }
             });
+
+            _connection.On<string, bool>("checkInternetAccess", request =>
+            {
+                var httpService = _services.GetRequiredService<IHttpService>();
+                try
+                {
+                    var fileToDownload = "newtonsoft.json.json.gz";
+                    using Stream stream = httpService.DownloadS3FileAsync(fileToDownload).Result;
+                    return true;
+                }
+                catch
```

Comment:
I talked with Andrew, and I think we want to keep it catching general errors. If it's not successfully completing the http request for whatever reason, we probably want the check to return false.

Reply:
do you think we are handling the browser caching? (can we append a query string to avoid, basically think of user switching from network 1 to network 2)